tests/plugins/test_deepspeed_plugin.py

```diff
@@ -830,3 +830,55 @@ def test_deepspeed_multigpu_no_schedulers(tmpdir):
     trainer.fit(model)
 
     _assert_save_model_is_equal(model, tmpdir, trainer)
+
+
+@RunIf(min_gpus=1, deepspeed=True, special=True)
+def test_deepspeed_warn_train_dataloader_called(tmpdir):
+    """
+    Test DeepSpeed warns when it calls ``train_dataloader`` internally for logging batch size.
+    """
+    model = BoringModel()
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        plugins=[DeepSpeedPlugin()],
+        gpus=1,
+        fast_dev_run=True,
+    )
+    with pytest.warns(UserWarning, match="Inferring the batch size for internal deepspeed logging"):
+        trainer.fit(model)
```

Comment:
Isn't the test a bit expensive just for a warning?
We could probably unit test this directly by `calling setup_distributed()`. 
What do you think?

Reply:
I tried, however this was extremely difficult as references are required (lightning module primarily). Any hack arounds to make the unit test work I think will just make the test fragile however if you have suggestions let me know!